entity-framework/core/index.md

```diff
@@ -1,39 +1,61 @@
 
-## The Model
+## The model
+
+With EF Core, data access is performed using a model. A model is made up of entity classes and a context object that represents a session with the database. The context object allows querying and saving data. For more information, see [Creating a Model](xref:core/modeling/index).
 
-With EF Core, data access is performed using a model. A model is made up of entity classes and a context object that represents a session with the database, allowing you to query and save data. See [Creating a Model](xref:core/modeling/index) to learn more.
+EF supports the following model development approaches:
 
-You can generate a model from an existing database, hand code a model to match your database, or use [EF Migrations](xref:core/managing-schemas/migrations/index) to create a database from your model, and then evolve it as your model changes over time.
+* Generate a model from an existing database.
+* Hand code a model to match the database.
+* Use [EF Migrations](xref:core/managing-schemas/migrations/index) to create a database from a model. Migrations allow evolving the database as the model changes.
 
 [!code-csharp[Main](../../samples/core/Intro/Model.cs)]
 
 ## Querying
 
-Instances of your entity classes are retrieved from the database using Language Integrated Query (LINQ). See [Querying Data](xref:core/querying/index) to learn more.
+Instances of your entity classes are retrieved from the database using Language [Integrated Query (LINQ)](/dotnet/csharp/programming-guide/concepts/linq/). For more information, see [Querying Data](xref:core/querying/index).
 
 [!code-csharp[Main](../../samples/core/Intro/Program.cs#Querying)]
 
-## Saving Data
+## Saving data
 
 Data is created, deleted, and modified in the database using instances of your entity classes. See [Saving Data](xref:core/saving/index) to learn more.
 
 [!code-csharp[Main](../../samples/core/Intro/Program.cs#SavingData)]
 
+## EF ORM considerations
+
+While EF Core is good at abstracting many programming details, there are some best practices applicable to any ORM that help to avoid common pitfalls in production apps:
+
+ - Intermediate-level knowledge or higher of the underlying database server is essential to architect, debug, profile, and migrate data in high performance production apps. For example, knowledge of primary and foreign keys, constraints, indexes, normalization, DML and DDL statements, data types, profiling, etc.
+- Functional and integration testing:  It's important to replicate the production environment as close as possible to:
+  - Find issues in the app that only show up when using a specific versions or edition of the database server .
+  - Catch breaking changes when upgrading EF Core and other dependencies. For example, adding or upgrading frameworks like ASP.NET Core, Application Insights, Serilog, or Automapper. These dependencies can affect EF Core in unexpected ways.
```

Comment:
Definitely ASP.NET and Automapper (and e.g. OData), but App Insights and Serilog?